src/device/usbd.c

```diff
@@ -40,17 +40,27 @@
 //--------------------------------------------------------------------+
 // Device Data
 //--------------------------------------------------------------------+
+
+enum
+{
+  STATE_POWERED,
+  STATE_DEFAULT,

```

Comment:
Is there a need for an "attached" state? I'm not sure how it would change the rest of the code, though, perhaps goes from attached to default after a reset?